src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2689,27 +2689,22 @@ exprt c_typecheck_baset::do_special_functions(
       throw 0;
     }
 
-    side_effect_expr_function_callt try_constant{expr};
-    typecheck_function_call_arguments(try_constant);
-    exprt argument = try_constant.arguments().front();
-    simplify(argument, *this);
-    const auto int_constant = numeric_cast<mp_integer>(argument);
+    typecheck_function_call_arguments(expr);
+
+    exprt clz =
+      count_leading_zeros_exprt{expr.arguments().front(),
+                                has_prefix(id2string(identifier), "__lzcnt"),
+                                expr.type()};
+    clz.add_source_location() = source_location;
 
     if(
-      !int_constant.has_value() || *int_constant == 0 ||
-      argument.type().id() != ID_unsignedbv)
+      config.ansi_c.mode == configt::ansi_ct::flavourt::GCC ||
+      config.ansi_c.mode == configt::ansi_ct::flavourt::CLANG)
     {
-      return nil_exprt{};
+      simplify(clz, *this);
     }
```

Comment:
I wouldn't simplify these up-front -- the simplifier is called by the front-end whenever a constant is needed.

Reply:
Ah, true, will fix.